messagingpush/src/main/java/io/customer/messagingpush/CustomerIOPushNotificationHandler.kt

```diff
@@ -139,22 +142,44 @@ internal class CustomerIOPushNotificationHandler(private val remoteMessage: Remo
         }
 
         // set pending intent
-        val pushContentIntent = Intent(CustomerIOPushReceiver.ACTION)
-        pushContentIntent.setClass(context, CustomerIOPushReceiver::class.java)
-
-        pushContentIntent.putExtras(bundle)
-        val notificationClickedIntent = PendingIntent.getBroadcast(
+        createIntentFromLink(
             context,
             requestCode,
-            pushContentIntent,
-            flags
-        )
-        notificationBuilder.setContentIntent(notificationClickedIntent)
+            bundle.getString(DEEP_LINK_KEY)
+        )?.let { pendingIntent ->
+            notificationBuilder.setContentIntent(pendingIntent)
+        }
 
         val notification = notificationBuilder.build()
         notificationManager.notify(requestCode, notification)
     }
 
+    private fun createIntentFromLink(
+        context: Context,
+        requestCode: Int,
+        deepLink: String?
+    ): PendingIntent? {
+        // In Android 12, you must specify the mutability of each PendingIntent
+        val flags = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            PendingIntent.FLAG_IMMUTABLE or PendingIntent.FLAG_UPDATE_CURRENT
+        } else {
+            PendingIntent.FLAG_UPDATE_CURRENT
+        }
+
+        val pushContentIntents = sdkConfig.urlHandler?.createIntentsForLink(deepLink)
+            ?: deepLinkUtil.createIntentsForLink(context, deepLink)
```

Comment:
```suggestion
        val pushContentIntents = sdkConfig.urlHandler?.createIntentsForLink(deepLink)
            ?: deepLinkUtil.createDefaultDeepLinkHandlerIntent(context, deepLink) ?: createDefaultOpenAppIntent(context)
```

I think something like this reads better. The code right now reads like we only support deep links but if a push doesn't have a deep link in it, the SDK can just launch an Activity.